pkgs/test/by-name/check-references/README.md

```diff
@@ -0,0 +1,3 @@
+Usage: nix-reference-check [FILES]
+
+Checks that all given Nix files don't transitively and statically refer to files outside their own directory
```

Comment:
```suggestion
Checks that all given Nix files don't transitively and statically refer to files outside the directory which contains them.

Dynamic references, produced by using `+` to concatenate a path with a string, are not checked.  For example, this tool will not prevent:

  import (./. + "/../../../default.nix")
```

Reply:
I tried this test case and `check-references` allowed it.  IIUC this is intended, since the tool only does what can be done without having to evaluate anything.

Reply:
Oh yeah that's a good point. I'm not sure to what extent these exact semantics should be specified though. It's not really important for contributors to know the exact rules, and if they're interested, they could check the code or try it out. And I'd hope that reviewers catch people doing such workarounds.